### ISSUE TITLE
[ONNX] RuntimeError: THPVariable_Check(tuple_elem) INTERNAL ASSERT FAILED at "/pytorch/torch/csrc/jit/passes/onnx/shape_type_inference.cpp":676,

### ISSUE THREAD

**Issue by Ki6an** [pytorch/pytorch#53730]
## 🐛 Bug
hey, in `torch version 1.7.0`, I was successfully able to export the T5 decoder to onnx. as shown below.
```python
        _ = torch.onnx.export(decoder_with_lm_head_init,
                            (input_ids_dec, attention_mask, enc_out),
                            f"{output_prefix}-decoder-with-lm-head_initial.onnx",
                            export_params=True,
                            opset_version=12,
                            input_names=['input_ids', 'encoder_attention_mask', 'encoder_hidden_states'],
                            output_names=['logits', 'past_key_values'],
                            dynamic_axes={
                              'input_ids': {0:'batch', 1: 'sequence'}, # batch_size, seq_length = input_shape
                              'encoder_hidden_states': {0:'batch', 1: 'sequence'},
                              'logits': {0:'batch', 1: 'sequence'},
                              "past_key_values" : {0:'batch', 1: 'sequence'},
                              'encoder_attention_mask' : {0:'batch', 1: 'sequence'},
                            } 
                            )
``` 
but, since I updated torch to `1.8.0` 
I'm facing this weird issue...
```python
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-28-baab8a36e37d> in <module>()
----> 1 generate_onnx_representation(model_to_be_converted, 't5')

4 frames
<ipython-input-26-b84fa7abe03a> in generate_onnx_representation(pretrained_version, output_prefix, model)
    152                               'logits': {0:'batch', 1: 'sequence'},
    153                               "past_key_values" : {0:'batch', 1: 'sequence'},
--> 154                               'encoder_attention_mask' : {0:'batch', 1: 'sequence'},
    155                             } 
    156                             )

/usr/local/lib/python3.7/dist-packages/torch/onnx/__init__.py in export(model, args, f, export_params, verbose, training, input_names, output_names, aten, export_raw_ir, operator_export_type, opset_version, _retain_param_name, do_constant_folding, example_outputs, strip_doc_string, dynamic_axes, keep_initializers_as_inputs, custom_opsets, enable_onnx_checker, use_external_data_format)
    274                         do_constant_folding, example_outputs,
    275                         strip_doc_string, dynamic_axes, keep_initializers_as_inputs,
--> 276                         custom_opsets, enable_onnx_checker, use_external_data_format)
    277 
    278 

/usr/local/lib/python3.7/dist-packages/torch/onnx/utils.py in export(model, args, f, export_params, verbose, training, input_names, output_names, aten, export_raw_ir, operator_export_type, opset_version, _retain_param_name, do_constant_folding, example_outputs, strip_doc_string, dynamic_axes, keep_initializers_as_inputs, custom_opsets, enable_onnx_checker, use_external_data_format)
     92             dynamic_axes=dynamic_axes, keep_initializers_as_inputs=keep_initializers_as_inputs,
     93             custom_opsets=custom_opsets, enable_onnx_checker=enable_onnx_checker,
---> 94             use_external_data_format=use_external_data_format)
     95 
     96 

/usr/local/lib/python3.7/dist-packages/torch/onnx/utils.py in _export(model, args, f, export_params, verbose, training, input_names, output_names, operator_export_type, export_type, example_outputs, opset_version, _retain_param_name, do_constant_folding, strip_doc_string, dynamic_axes, keep_initializers_as_inputs, fixed_batch_size, custom_opsets, add_node_names, enable_onnx_checker, use_external_data_format, onnx_shape_inference, use_new_jit_passes)
    696                                 training=training,
    697                                 use_new_jit_passes=use_new_jit_passes,
--> 698                                 dynamic_axes=dynamic_axes)
    699 
    700             # TODO: Don't allocate a in-memory string for the protobuf

/usr/local/lib/python3.7/dist-packages/torch/onnx/utils.py in _model_to_graph(model, args, verbose, input_names, output_names, operator_export_type, example_outputs, _retain_param_name, do_constant_folding, _disable_torch_constant_prop, fixed_batch_size, training, use_new_jit_passes, dynamic_axes)
    483 
    484         output_tensors, out_desc = torch._C._jit_flatten(tuple(output_wrapped))
--> 485         torch._C._jit_pass_onnx_assign_output_shape(graph, output_tensors, out_desc, _onnx_shape_inference)
    486 
    487     _set_input_and_output_names(graph, input_names, output_names)

RuntimeError: THPVariable_Check(tuple_elem) INTERNAL ASSERT FAILED at "/pytorch/torch/csrc/jit/passes/onnx/shape_type_inference.cpp":676, please report a bug to PyTorch. 
```

**Collecting environment information...**

PyTorch version: 1.8.0+cu101
Is debug build: False
CUDA used to build PyTorch: 10.1
ROCM used to build PyTorch: N/A

OS: Ubuntu 18.04.5 LTS (x86_64)
GCC version: (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
Clang version: 6.0.0-1ubuntu2 (tags/RELEASE_600/final)
CMake version: version 3.12.0

Python version: 3.7 (64-bit runtime)
Is CUDA available: False
CUDA runtime version: 11.0.221
GPU models and configuration: Could not collect
Nvidia driver version: Could not collect
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.7.6.5
/usr/lib/x86_64-linux-gnu/libcudnn.so.8.0.4
/usr/lib/x86_64-linux-gnu/libcudnn_adv_infer.so.8.0.4
/usr/lib/x86_64-linux-gnu/libcudnn_adv_train.so.8.0.4
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_infer.so.8.0.4
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_train.so.8.0.4
/usr/lib/x86_64-linux-gnu/libcudnn_ops_infer.so.8.0.4
/usr/lib/x86_64-linux-gnu/libcudnn_ops_train.so.8.0.4
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] numpy==1.19.5
[pip3] torch==1.8.0+cu101
[pip3] torchsummary==1.5.1
[pip3] torchtext==0.9.0
[pip3] torchvision==0.9.0+cu101
[conda] Could not collect

cc @ezyang @gchanan @zou3519 @bdhirsh @jbschlosser @anjali411 @houseroad @spandantiwari @lara-hdr @BowenBao @neginraoof @SplitInfinity @gmagogsfm

**Comment by gchanan** [comment]
do you have a self-contained reproduction?

**Comment by BowenBao** [comment]
Likely fixed by #53311

**Comment by malfet** [comment]
@BowenBao can you please validate the PR fixes the issue? Also #53311 does not apply cleanly against 1.8 branch

**Comment by BowenBao** [comment]
Verified locally that this can be repro-ed with 1.8.0, and resolved with cherry-picking #53311. Closing.
